shell/src/main/java/org/apache/accumulo/shell/commands/CompactCommand.java

```diff
@@ -147,30 +170,28 @@ public int execute(final String fullCommand, final CommandLine cl, final Shell s
       compactionConfig.setIterators(new ArrayList<>(iterators));
     }
 
-    Map<String,String> selectorOpts = new HashMap<>();
-    Map<String,String> configurerOpts = new HashMap<>();
-    getConfigurableCompactionStrategyOpts(cl, selectorOpts, configurerOpts);
+    setupConfigurableCompaction(cl, compactionConfig);
 
     if (cl.hasOption(strategyOpt.getOpt())) {
-      if (!selectorOpts.isEmpty() || !configurerOpts.isEmpty())
+      if (cl.hasOption(selectorOpt.getLongOpt()) || cl.hasOption(configurerOpt.getLongOpt())) {
         throw new IllegalArgumentException(
-            "Can not specify compaction strategy with file selection and file output options.");
-
+            "Can not specify a strategy with a selector or configurer");
+      }
       configureCompactionStrat(cl);
     }
 
-    if (!selectorOpts.isEmpty()) {
-      PluginConfig selectorCfg = new PluginConfig(
-          "org.apache.accumulo.tserver.compaction.strategies.ConfigurableCompactionStrategy",
-          selectorOpts);
-      compactionConfig.setSelector(selectorCfg);
+    if (cl.hasOption(selectorOpt.getLongOpt())) {
```

Comment:
If the new options can't be set along with a strategy, shouldn't these two if statements be in an else?

Reply:
Fixed in abc961a